### PR TITLE
Performance & correctness fixes: O(L²)→O(L×K) distances/masks, seed=0 bug, symmetry bias, deduplication

### PR DESCRIPTION
**Pull request by mooreneural**
A batch of performance and correctness fixes found during a codebase audit.
No model architecture or hyperparameters are changed; all pretrained checkpoints
load and produce identical output (except where a correctness bug is fixed).

## Correctness fixes

- **Symmetry bias bug** (`model_utils.py` `sample()`): The symmetry path accumulated
  weighted logits correctly but applied only the *last* group member's `bias[:, t]`
  to the softmax. Fixed to accumulate a weighted `total_bias` across all group members.

- **`--seed 0` silently ignored** (`run.py`): `if args.seed:` treats `0` as falsy,
  falling back to a random seed. Changed to `if args.seed is not None:` and the
  argparse default from `0` to `None`.

- **Bare `except:`** (`data_utils.py`): Swallowed `KeyboardInterrupt` and `SystemExit`.
  Changed to `except Exception:`.

### O(L²) → O(L×K) distance features (`_get_rbf`)

`_get_rbf` built a full `[B, L, L]` pairwise distance matrix and discarded all but K
entries. Since k-NN indices `E_idx` are already known, we `gather_nodes` neighbor
coordinates first and compute only the K needed distances per residue. Applied to all
three feature classes (`ProteinFeaturesLigand`, `ProteinFeatures`,
`ProteinFeaturesMembrane`).

### O(L²) → O(L×K) causal mask

`sample()`, `score()`, and `single_aa_score()` constructed a full `[B, L, L]`
permutation matrix and ran an `O(L³)` einsum to determine which neighbours were decoded
before each position. Replaced with `rank = argsort(decoding_order)`, gather neighbour
ranks, scalar compare, same result at `O(L×K)`.

### CPU benchmark (`python benchmark.py`, B=1, K=32, num_rbf=16)

**`_get_rbf` - pairwise distance RBF features**

| L | before (ms) | after (ms) | speedup |
|--:|--:|--:|--:|
| 50 | 0.22 | 0.17 | 1.2x |
| 100 | 0.60 | 0.27 | 2.2x |
| 200 | 1.82 | 0.58 | 3.1x |
| 400 | 6.21 | 0.71 | 8.7x |
| 600 | 16.44 | 1.15 | 14.3x |

**Causal mask - backward/forward neighbour masks for decoding**

| L | before (ms) | after (ms) | speedup |
|--:|--:|--:|--:|
| 50 | 0.19 | 0.04 | 4.7x |
| 100 | 0.18 | 0.04 | 5.0x |
| 200 | 0.35 | 0.06 | 5.3x |
| 400 | 0.71 | 0.07 | 9.8x |
| 600 | 1.53 | 0.10 | 15.1x |

These are called per-residue in the autoregressive loop, so real savings multiply by L.

### Other performance changes

- **RBF µ pre-allocated as buffer**: `_rbf` called `torch.linspace(2, 22, 16)` on
  every forward pass. Moved to a non-persistent `register_buffer` (moves with
  `.to(device)`, not saved to `state_dict` so checkpoints remain compatible).
- **OpenFold lookup tables pre-converted** (`sc_utils.py`): Four NumPy arrays were
  passed to `torch.tensor()` inside the denoising loop on every iteration.
- **14×14 sidechain RBF vectorized** (`sc_utils.py` `features_decode`): 196
  sequential `_get_rbf` calls replaced with a single batched `[B, L, 14, K, 14]`
  distance tensor. Feature order preserved so pretrained weights are unaffected.
  Primary win is on GPU (196 kernel launches → 1 fused op).

## Cleanup

- **Duplicate definitions removed** (`data_utils.py`): A 167-line block inside
  `parse_PDB` redefined `element_list`, `element_dict`, `restype_3to1`, and
  `restype_STRtoINT` locally, shadowing identical module-level definitions.
- **O(N) → O(1) dict lookup** (`data_utils.py` `get_aligned_coordinates`):
  `if code in list(CA_dict)` converted the dict to a list on every residue check.

## Reproducing the benchmark

```bash
python benchmark.py             # CPU
python benchmark.py --device cuda   # GPU (shows larger gains from fused kernels)

### CODE REVIEW

**Comment by mooreneural** [comment]
**Real-PDB benchmark** on the three bundled input structures, GPU (CUDA 12.8, PyTorch 2.11):

```bash
python benchmark_pdb.py --device cuda
```

```text
Device: cuda   K(mpnn)=32  K(sc)=48
Median of 20 runs (ms), 5 warmup runs.

PDB            L |  rbf before  rbf after  speedup |  mask before  mask after  speedup |  sc before  sc after  speedup
----------------------------------------------------------------------------------------------------------------------
1BC8          93 |        0.13       0.14     0.9x |         0.24        0.14     1.7x |      32.92      1.11    29.7x
2GFB        3464 |       10.10       0.16    63.5x |         4.91        0.11    46.0x |      37.08     43.64     0.8x
4GYT         354 |        0.13       0.15     0.9x |         0.24        0.10     2.4x |      30.63      4.34     7.1x
```

Gains scale with sequence length as expected. 2GFB (3,464 residues, large multimer) shows 63× on `_get_rbf` and 46× on the causal mask, the structures where O(L²) hurts most in practice. The 14×14 sidechain RBF vectorization favours smaller proteins on GPU (29.7× on 1BC8) but regresses at very large L where the fused tensor exceeds L2 cache; the 196 sequential kernel launches are more cache-friendly there.

No checkpoint required to reproduce - `benchmark_pdb.py` parses ATOM records directly.
